### ISSUE TITLE
resizing the side panel doesn't work

### ISSUE THREAD

**Issue by nicebyte** [rxi/lite#113]
**steps to reproduce**: open the editor, hover mouse over the border between main edit area and sidebar, click and drag
**expected behavior**: side panel changes width
**observed behavior**: nothing happens

**demonstration**:
![lite-resize](https://user-images.githubusercontent.com/12521759/82632251-89fd9900-9bac-11ea-9680-10fd5032aa04.gif)

**Comment by harikt** [comment]
Seems similar to https://github.com/rxi/lite/issues/23

**Comment by harikt** [comment]
This is what I did 

```
config.treeview_size = 250 * SCALE
```

I don't think we can resize it as mentioned in issue 23.

**Comment by nicebyte** [comment]
in that case can we consider this issue a feature request to implement resizing the sidebar with a mouse? having to modify a config file every time i want to resize the sidebar is not very convenient.

**Comment by harikt** [comment]
@nicebyte I agree with you. Because I have also similar issue with the width. But what I try is increasing the width, and use `ctrl+\` to hide it.

**Comment by andsve** [comment]
In `treeview.lua:190`;
```
node:split("left", view, false)
```

`false` here corresponds to `locked`, changing it to `true` will make the side panel/treeview resizeable. However, it will not use the `treeview_size` directly, but split the root view equally. Might be a good point to start investigating a solution, let me know if you come up with something. :)

**Comment by rxi** [comment]
@andsve two small things I noticed:
* With how the override on `on_mouse_moved` is done — where the original function isn't called — that other plugins that patch into the same function stop working, for example this breaks the `linter` plugin
* How the `background` color is set means that changing the theme doesn't update the background color. In other plugins that use fall-back colors I'd do: ```style.minimap_background_color or style.background_color``` at the usage point (the color would be expected in the `style` table too)

**Comment by rxi** [comment]
As mentioned before this is intended behaviour, that being said supporting this functionality is a valid feature request (thus this issue will remain open). The thing that's preventing this isn't the treeview in general but the UI layout system as a whole -- that is, I'd rather not add a hack to fix this, and instead would prefer changing how the `RootView` works to better support this. I haven't yet settled on an elegant solution of how this can be resolved.
